lib/services/android-project-service.ts

```diff
@@ -29,24 +29,30 @@ class AndroidProjectService implements IPlatformProjectService {
 		private $propertiesParser: IPropertiesParser) {
 	}
 
+	private _platformData: IPlatformData = null;
 	public get platformData(): IPlatformData {
-		var projectRoot = path.join(this.$projectData.platformsDir, "android");
-
-		return {
-			frameworkPackageName: "tns-android",
-			normalizedPlatformName: "Android",
-			appDestinationDirectoryPath: path.join(projectRoot, "assets"),
-			appResourcesDestinationDirectoryPath: path.join(projectRoot, "res"),
-			platformProjectService: this,
-			emulatorServices: this.$androidEmulatorServices,
-			projectRoot: projectRoot,
-			deviceBuildOutputPath: path.join(this.$projectData.platformsDir, "android", "bin"),
-			validPackageNamesForDevice: [
-				util.format("%s-%s.%s", this.$projectData.projectName, "debug", "apk"),
-				util.format("%s-%s.%s", this.$projectData.projectName, "release", "apk")
-			],
-			frameworkFilesExtensions: [".jar", ".dat", ".so"]
-		};
+		if (!this._platformData) {
```

Comment:
Because this.platformData is being called quite many times. While string operations are not the slowest ones, creating new objects is a bit taxing in V8.